### ISSUE TITLE
Unable to shift focus to back button on nested menu via keyboard

### ISSUE THREAD

**Issue by bjankord** [cerner/terra-core#2082]
# Bug Report

## Description
When opening a nested menu and shifting focus away from the back button, the user is unable to shift focus back to the back button with a keyboard.

## Steps to Reproduce

1. Navigate to https://engineering.cerner.com/terra-core/#/components/terra-menu/menu/menu
2. Click on first menu example
3. Click on Nested Menu Item
4. Press Tab Key
5. Press Shift+Tab keys
6. Focus does not shift back to the nested menu back button
7. Press Tab key to try and shift focus, focus does not shift from the first menu item

## Additional Context / Screenshots
Shifting focus to the back button kinda works on [this example](https://engineering.cerner.com/terra-core/#/raw/tests/terra-menu/menu/menu/sub-menu).

The user can cycle through focusing on the back button and the menu item pressing tab, but shift+tab does not shift focus back from the menu item.

## Expected Behavior
Users can use a keyboard to shift focus to back button in the nested menu

## Environment

* Component Name and Version: Seen in terra-menu v3.28.0
* Browser Name and Version: Chrome Version 70

**Comment by supreethmr** [comment]
Along with unable shift focus back on click of shift + tab on nested menu I have observed that focus does not shift from the first menu item on click of tab. it can only move with the down arrow keys @neilpfeiffer I would like to know if this is the intended behavior of menu on Tab Key press. 

if this not intended behavior Then this issue is due to event.preventDefault() that is added in onKeyDown event of menu item https://github.com/cerner/terra-core/blob/68ebbf11da8170926d699be21d5d2b504b09b5c9/packages/terra-menu/src/_MenuContent.jsx#L207. 

Since it blocks default behavior of keys pressed navigation was not happening on press of Tab and shift + Tab. 

Hence we can have condition to skip preventDefault for Tab and Shift + Tab like below :

const shiftTabClicked = (event.shiftKey && event.nativeEvent.keyCode === MenuUtils.KEYCODES.TAB);
const tabClicked = (event.nativeEvent.keyCode === MenuUtils.KEYCODES.TAB);
if (!(shiftTabClicked || tabClicked)) {
event.preventDefault(); // Prevent Default will not be invoked on click of tab and shift + tab
}
